File: tdd-01/code/java/src/com/awl/tdd/test/FakeEngine.java
package com.awl.tdd.test;

import com.awl.tdd.model.Engine;

public class FakeEngine extends Engine {

	private boolean startCalled;
	private String fakeSound;

	public FakeEngine(String fakeSound) {
		this.fakeSound = fakeSound;
	}

	public String start() {
		startCalled = true;
		return fakeSound;
	}

	public boolean isStartCalled() {
		return startCalled;
	}
	
}

File: tdd-01/code/objectivec/TDD-Car-01/test/FakeEngine.m
//
//  FakeEngine.m
//  TDD-Car-01
//
//  Copyright (c) 2014 Company. All rights reserved.
//

#import "FakeEngine.h"

@implementation FakeEngine

- (id)initWithSound:(NSString *)sound_ {
    
    if ( self = [super init] ) {
        self->sound = sound_;
        return self;
        
    } else
        
        return nil;
    
}

- (NSString*) start {
	self->startCalled = true;
    return self->sound;
}

- (bool) isStartCalled {
    return startCalled;
}

@end
